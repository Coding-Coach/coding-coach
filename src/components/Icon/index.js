import React from 'react';
import { any, func, number, string, object } from 'prop-types';

export default function Icon(props) {
  const {
    children,
    className,
    color,
    title,
    height,
    onClick,
    style,
    width,
    vectors,
    viewBox,
  } = props;

  return (
    <svg
      width={width}
      height={height}
      viewBox={viewBox}
      aria-labelledby="title"
      className={className}
      style={style}
      onClick={onClick}
    >
      <title id="title">{title}</title>
      {vectors && <path fill={color} d={vectors} />}
      {children}
    </svg>
  );
}

Icon.propTypes = {
  children: any,
  className: string,
  color: string,
  title: string,
  height: number,
  onClick: func,
  style: object,
  vectors: string,
  viewBox: string,
  width: number,
};
