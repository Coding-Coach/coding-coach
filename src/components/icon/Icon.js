import React from 'react';
import { any, number, string, object } from 'prop-types';

export default function Icon({
  children,
  className,
  color,
  title,
  height,
  style,
  width,
  vectors,
  viewBox,
}) {
  return (
    <svg
      width={width}
      height={height}
      viewBox={viewBox}
      aria-labelledby="title"
      className={className}
      style={style}
    >
      <title id="title">{title}</title>
      {vectors && <path fill={color} d={vectors} />}
      {children}
    </svg>
  );
}

Icon.propTypes = {
  children: any,
  className: string,
  color: string,
  title: string,
  height: number,
  style: object,
  vectors: string,
  viewBox: string,
  width: number,
};
