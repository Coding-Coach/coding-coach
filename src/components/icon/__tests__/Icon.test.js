import React from 'react';
import { render } from 'react-testing-library';
import Icon from '../Icon';

describe('components/icon/Icon', () => {
  it('should render the icon', () => {
    const { container } = render(<Icon />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should set the title', () => {
    const { getByText } = render(<Icon title="testing" />);

    expect(getByText('testing')).toBeTruthy();
  });

  it('should render vectors', () => {
    const { container } = render(<Icon vectors="123" />);

    expect(container.querySelector('path[d="123"]')).toBeTruthy();
  });
});
