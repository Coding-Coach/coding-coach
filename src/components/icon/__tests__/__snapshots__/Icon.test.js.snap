// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/icon/Icon should render the icon 1`] = `
<svg
  aria-labelledby="title"
>
  <title
    id="title"
  />
</svg>
`;
